UploadRequestHandler: Support resource sanitation

lardser opened the pull request:
By providing corresponding implementations of
IResourceSanitizerImplementor, files uploaded
via the UploadRequestHandler can be sanitized.

The XmlUtility provides now a method to create
a namespace aware xml document from a raw string
representation of a xml.

434705

lardser committed:
UploadRequestHandler: Support resource sanitation

By providing corresponding implementations of
IResourceSanitizerImplementor, files uploaded
via the UploadRequestHandler can be sanitized.

The XmlUtility provides now a method to create
a namespace aware xml document from a raw string
representation of a xml.

434705

awegmueller commented on org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/XmlUtility.java:
```diff
@@ -492,8 +505,24 @@ public static Document getXmlDocument(URL url) {
+  }
+
+  /**
+   * Creates an xml {@link Document} filled with the content of the given {@link String}.
```
"xml" should be uppercase in regular texts (multiple occurrences in this javadoc).

awegmueller commented on org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/XmlUtility.java:
```diff
@@ -492,8 +505,24 @@ public static Document getXmlDocument(URL url) {
+   * @throws ProcessingException
+   *     if there is an error parsing the content of the {@link String} into a {@link Document}.
+   */
+  public static Document getXmlDocument(String rawXml, boolean namespaceAware) {
```
Please add a unit test in XmlUtilityTest for the new methods.

lardser committed:
UploadRequestHandler: Support resource sanitation

By providing corresponding implementations of
IResourceSanitizerImplementor, files uploaded
via the UploadRequestHandler can be sanitized.

The XmlUtility provides now a method to create
a namespace aware xml document from a raw string
representation of a xml.

434705
